File: C#1/Homeworks/03-Operators-and-Expressions-Homework/07PointInCircle/PointInCircle.cs
using System;
using System.Threading;
using System.Globalization;

//Write a program that reads the coordinates of a point x and y and 
//using an expression checks if given point (x, y) is inside a circle  

//K({0, 0}, 2)  - the center has coordinates (0, 0) and the circle has radius 2.

//The program should then print "yes DISTANCE" if the point is inside the circle 
//or "no DISTANCE" if the point is outside the circle.
//In place of DISTANCE print the distance from the beginning of the 
//coordinate system - (0, 0) - to the given point.

//input : 
//exactly two lines, the first containing the x coordinate, 
//the second containing the y coordinate.

//output :
//single line in the format described above. 
//The distance should be always printed with 2-digit precision after the 
//floating point.


    class PointInCircle
    {
        static void Main()
        {
            Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;

            float radius = 2f;
            float xc = 0.0f;
            float yc = 0.0f;

            float x = float.Parse(Console.ReadLine());
            float y = float.Parse(Console.ReadLine());
                       
            //float dx = Math.Abs(x - xc); -> dx = x
            //float dy = Math.Abs(y - yc); -> dy = y
            //double distance = Math.Sqrt((dx * dx) + (dy * dy));

            double distance = Math.Sqrt((x * x) + (y * y));

            bool isInside = (distance <= radius) ? true : false;

            if (isInside) 
            { 
                Console.Write("yes "); 
            }
            else
            { 
                Console.Write("no ");  
            }
            Console.Write("{0:f2}\n",distance);
        }
    }

File: JS/HW/Operators-and-Expressions/pointInCircle.js
function solve(args){
    var x = +args[0];
    var y = +args[1];
    
    var radius = 2;
    
    //d=SQRT((xp−xc)2+(yp−yc)2)
    var distance = Math.sqrt(x*x + y*y);
     
    var inside = distance <= radius;
    if(inside){
       console.log("yes " + distance.toFixed(2)); 
    }else {
        console.log("no " + distance.toFixed(2));
    }
}


solve(['-2', '0']);
solve(['-1', '2']);
solve(['1.5', '-1']);
solve(['1.5', '-1']);
solve(['-1.5', '-1.5']);
solve(['100', '-30']);
solve(['0', '0']);
solve(['0.2', '-0.8']);
solve(['0.9', '-1.93']);
solve(['1', '1.655']);
solve(['0', '1']);